test/docker-e2e/e2e_state_sync_test.go

```diff
@@ -214,3 +210,423 @@ func (s *CelestiaTestSuite) TestStateSyncMocha() {
+
+	stateSyncClient, err := stateSyncNode.GetRPCClient()
+	s.Require().NoError(err, "failed to get state sync client")
+
+	t.Log("Phase 4: Waiting for state sync completion and validation")
+	heightHistory := s.monitorSyncProgress(ctx, t, stateSyncClient, latestHeight, stateSyncTimeout)
+
+	// Verify that state sync was used (not block sync)
+	dockerNode := stateSyncNode.(*tastoradockertypes.ChainNode)
+	verifySyncMethod(t, heightHistory, dockerNode)
+
+	// Validate: /status shows catching_up=false
+	status, err := stateSyncClient.Status(ctx)
+	s.Require().NoError(err, "failed to get status from state sync node")
+	s.Require().False(status.SyncInfo.CatchingUp, "state sync node should not be catching up")
+	t.Logf("State sync node status: catching_up=%t, height=%d",
+		status.SyncInfo.CatchingUp, status.SyncInfo.LatestBlockHeight)
+
+	// get fresh chain height for accurate comparison (chain continued producing blocks during state sync)
+	currentChainHeight, err := s.GetLatestBlockHeight(ctx, rpcClient)
+	s.Require().NoError(err, "failed to get current chain height")
+
+	// Verify that the node actually performed state sync by checking it reached current height
+	nodeHeight := status.SyncInfo.LatestBlockHeight
+	s.Require().GreaterOrEqual(nodeHeight, currentChainHeight-5,
+		"state sync node should be within 5 blocks of the current chain tip")
+	s.Require().LessOrEqual(nodeHeight, currentChainHeight+2,
+		"state sync node should not be significantly ahead of the chain")
+	t.Logf("State sync successful: node reached height %d (current chain tip: %d)",
+		nodeHeight, currentChainHeight)
+
+	// Validate: ABCIInfo.app_version == targetAppVersion
+	syncedAbciInfo, err := stateSyncClient.ABCIInfo(ctx)
+	s.Require().NoError(err, "failed to fetch ABCI info from state sync node")
+	s.Require().Equal(targetAppVersion, syncedAbciInfo.Response.GetAppVersion(),
+		"state sync node should have app version %d", targetAppVersion)
+	t.Logf("State sync node app version: %d", syncedAbciInfo.Response.GetAppVersion())
+
+	// Validate: Basic bank send succeeds from synced node
+	t.Log("Testing bank send transaction from state sync node")
+	testBankSend(s.T(), chain, cfg)
+	t.Log("Bank send transaction succeeded")
+
+	// Final liveness check
+	t.Log("Performing final liveness check")
+	s.Require().NoError(s.CheckLiveness(ctx, chain), "liveness check failed")
+	t.Log("Liveness check passed")
+}
+
+// checkSyncMetrics queries Prometheus metrics to determine sync method
+func checkSyncMetrics(t *testing.T, node *tastoradockertypes.ChainNode) (stateSync bool, blockSync bool, err error) {
+	ctx := context.Background()
+
+	// Try multiple approaches to get metrics
+	endpoints := []string{
+		"http://localhost:26660/metrics", // Default prometheus port
+		"http://127.0.0.1:26660/metrics", // Alternative localhost
+		"http://localhost:26657/metrics", // Alternative CometBFT port
+		"http://127.0.0.1:26657/metrics", // Alternative CometBFT port
```

Comment:
[question] are these really necessary? I assume tastora spins up the node with a reliable Prometheus endpoint so seems odd that we need to try all these.